### PR TITLE
Infrastructure: Add configuration to setup testing of Python code (pytest.ini + pyproject.toml only)

### PR DESCRIPTION
**Pull request by bernhardkaindl**
Only 2 config files are changed: This should be very easy to review.

@edwintorok @lindig 

Documentation is added at the configuration settings, so it should be very easy to use and maintain.

Interestingly, while this only adds useful configuration, it also helps coverage to gather the coverage of the new test actually calling the constructor of `class rrdd.Displatcher()` and its `__getattr__()` method to read an attribute, which the test indeed does:

https://app.codecov.io/gh/xapi-project/xen-api/commit/e3d21c2c225adafda56e33329be9e3b548984fb6/indirect-changes?flags%5B0%5D=python3.11

This means, without even intended for it, this already helps the test coverage results as an indirect change that you can examine at the Codecov.io link above.

### CODE REVIEW

**Review by psafont — APPROVED** [review]

File: pyproject.toml [hunk 1]
```diff
@@ -34,6 +34,68 @@ profile = "black"
...
+
+# -----------------------------------------------------------------------------
+# The source option specifies the directories to be measured:
+#
+source = ["scripts", "ocaml"]
+
+# -----------------------------------------------------------------------------
+# The omit option specifies files or directories to be omitted from the report:
+#
+omit = [  # These need some simple syntax update to be parseable:
```
Comment (Vincent-lau) [inline pyproject.toml:57]: what is going on with these scripts? I understand that this PR is more for infra change, do we have another PR that does the syntax updates for these?

File: pytest.ini [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+# testpaths: Directories to search for tests when no test targets are given.
+# scripts: Search ./scripts for tests, ocaml/xcp-rrdd: Search ./ocaml/xcp-rrdd
+testpaths =
+    scripts
+    ocaml/xcp-rrdd
+
+# norecursedirs: Directories to ignore when searching for tests, e.g. build
+# build: Ignore ./build when searching for tests:
+norecursedirs = build
```
Comment (Vincent-lau) [inline pytest.ini:53]: Is it `_build`?

File: pytest.ini [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+
+# addopts: Additional command line options for pytest, see pytest --help:
+# -rF: Show extra test summary info for failed tests:
+addopts = -rF
```
Comment (Vincent-lau) [inline pytest.ini:57]: my pytest gives me this. Is it `-rf`?
```
 -r chars              Show extra test summary info as specified by chars:
                        (f)ailed, (E)rror, (s)kipped, (x)failed, (X)passed,
                        (p)assed, (P)assed with output, (a)ll except passed
                        (p/P), or (A)ll. (w)arnings are enabled by default (see
                        --disable-warnings), 'N' can be used to reset the list.
                        (default: 'fE').

```
